### ISSUE TITLE
Collaborating on SA with SciPy

### ISSUE THREAD

**Issue by tupui** [SALib/SALib#463]
Following some discussion here https://github.com/SALib/SALib/issues/414#issuecomment-817680221_

**TL;DR** I am trying to get SA into SciPy, would you be interested in helping?

I recently had a chat with Sergei Kucherenko and he is interested to contribute SA to SciPy. So far he helped me with the Sobol' sequences and OC, SA is of prime interest for him and he is finding the perspective to have SA available in SciPy interesting.

There is not much left IMHO to quick-start real discussions about a SA inclusion in SciPy. As a major SA package in Python, your inputs would be valuable. To be clear, my long term goal would not be to absorb SALib, but more to propose a collaboration. It would be like SciPy's QMC module and PyQMC. We are discussing actively with Fred Hickernell. While there are some overlaps, PyQMC is OC more feature rich and state of the art, while SciPy proposes stable and foundational parts. They actually wrap SciPy's Sobol' generators. This way PyQMC can propose the latest algorithms to practitioners. If something proves to be great, then it can be proposed to SciPy for integration. The benefit is that due to SciPy's reach, we make our fields accessible to the most.

FYI, there are some discussions on SciPy's mailing list: https://mail.python.org/pipermail/scipy-dev/2021-April/024696.html There is even a suggestion to include SALib into SciPy if you would have some interest. It would be great if you or other maintainers could drop in and comment on the whole topic, this could help the discussion!

cc @ConnectedSystems @willu47

**Comment by chahank** [comment]
Not a developer from SALib, but I integrated SALib into the climate risk package CLIMADA. I would love to see good SA in SciPy to potentially have a maybe smaller version of what SALib currently contains, but maybe more stable.

**Comment by tupui** [comment]
> Not a developer from SALib, but I integrated SALib into the climate risk package CLIMADA. I would love to see good SA in SciPy to potentially have a maybe smaller version of what SALib currently contains, but maybe more stable.

Thanks @chahank. There are some movement on the mailing list, it's a heated discussion... (follow link above and here https://mail.python.org/pipermail/scipy-dev/2021-September/thread.html). Feel free to drop by!

**Comment by AdityaSavara** [comment]
I would certainly like it if Salib became a backend dependency of Scipy.

**Comment by tupui** [comment]
This happened and now Sobol' indices are available in SciPy (starting from 1.11). Thanks Takuya et al.!

https://scipy.github.io/devdocs/reference/generated/scipy.stats.sobol_indices.html

One possibility now would be to use SciPy's implementation in SALib. We added some mechanism that should allow us to get a similar feature set. And as usual if something is missing, please let me know and I am happy to have a look at how to improve SciPy's proposition 😃

**Comment by ConnectedSystems** [comment]
What do you think the likelihood of having a range of Sobol estimators in SciPy is @tupui ?

I know Saltelli 2010 is the default now, but not sure if other estimators have been added since?

**Comment by tupui** [comment]
This is a good question. I do want to add more methods (more expression for Sobol' indices, but also maybe other methods like moment independent, etc.)

Since there was quite "politics" at play, I want to wait at least until the release to propose more. I am planning to make a communication at the SciPy2023 conference in June as well (I will of course mention you and SALib!) and I hope this will generate wide interest. All that to support future improvement proposals 😃 

For now I created a tracking issue on my fork. Feel free to drop by and add anything. 

https://github.com/tupui/scipy/issues/14
